MekHQ/src/mekhq/campaign/universe/Systems.java

```diff
@@ -0,0 +1,500 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.universe;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.Writer;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+
+import org.joda.time.DateTime;
+import org.w3c.dom.DOMException;
+import org.w3c.dom.Node;
+
+import megamek.common.EquipmentType;
+import megamek.common.logging.LogLevel;
+import megamek.common.util.EncodeControl;
+import mekhq.MekHQ;
+import mekhq.MekHqXmlUtil;
+import mekhq.Utilities;
+import mekhq.campaign.universe.Planet.PlanetaryEvent;
+
+/** 
+ * This will eventually replace the Planets object as our source of system/planetary information
+ * using the new XML format that places planets within systems
+ * @author Taharqa
+ */
+
+public class Systems {
+    private final static Object LOADING_LOCK = new Object[0];
+    private static Systems systems;
+    
+    //private static ResourceBundle resourceMap = ResourceBundle.getBundle("mekhq.resources.Planets", new EncodeControl()); //$NON-NLS-1$
+    
+ // Marshaller / unmarshaller instances
+    private static Marshaller marshaller;
+    private static Unmarshaller unmarshaller;
+    static {
+        try {
+            JAXBContext context = JAXBContext.newInstance(LocalSystemList.class, Planet.class);
+            marshaller = context.createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_FRAGMENT, Boolean.TRUE);
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+            unmarshaller = context.createUnmarshaller();
+            // For debugging only!
+            // unmarshaller.setEventHandler(new javax.xml.bind.helpers.DefaultValidationEventHandler());
+        } catch(JAXBException e) {
+            MekHQ.getLogger().error(Systems.class, "<init>", e); //$NON-NLS-1$
+        }
+    }
+    
+    public static Systems getInstance() {
```

Comment:
Does this need to be synchronized somehow?

Reply:
@sixlettervariables Maybe? Its basically the same as Planets.getInstance() in the master branch which is not synchronized ATM.